File: data/code_pairs/cpp/1597.cpp
#include <iostream>
#include <fstream>
#include <iomanip>
#include <string>

void writeqnode(int myid, const double* qnode, int qnodesize) {
    std::string filename = std::to_string(1000 + myid) + ".txt";
    std::ofstream outFile(filename);

    if (!outFile.is_open()) {
        std::cerr << "Failed to open file: " << filename << std::endl;
        return;
    }

    outFile << std::scientific << std::setprecision(8);

    for (int i = 0; i < qnodesize; ++i) {
        outFile << " " << qnode[i] << std::endl;
    }

    outFile.close();
}

File: data/code_pairs/fortran/1597.f90
subroutine writeqnode(myid, qnode, qnodesize)
    implicit none
    integer :: i
    integer, intent(in) :: myid, qnodesize
    real*8, intent(in) :: qnode(qnodesize)
    character(len=20) :: filename

    write(filename, '(I4)') 1000+myid

    open(unit=1000+myid, file=trim(adjustl(filename)), status="replace", action="write")

    do i = 1, qnodesize
        write(1000+myid, '(1x, E14.8)') qnode(i)
    end do

    close(1000+myid)
end subroutine writeqnode